lib/Tumblr/StreamBuilder/FencepostRanking/FencepostCursor.php

```diff
@@ -122,9 +122,9 @@ public function get_head_offset(): int
     }
 
     /**
-     * @return StreamCursor
+     * @return StreamCursor|null
      */
-    public function get_tail_cursor(): StreamCursor
+    public function get_tail_cursor(): ?StreamCursor
```

Comment:
would this count as a breaking change that requires a new major version? probably.....

Reply:
this is just a change in the type-hinting, $tail_cursor can already be null

Reply:
but if it was ever null, we'd be seeing TypeErrors on our side, that's my point. code-wise, this could not return null before (even if we were doing it erroneously), and now it can, so that counts as a breaking change for any consumer of this code who was following it as intended. 

so i think we should just bump the major version number here.